_overviews/tutorials/scala-for-java-programmers.md

```diff
@@ -607,11 +969,41 @@ individual fields. The default implementation of `equals` is not
 usable, because as in Java it compares objects physically. We arrive
 at the following definition:
 
-    override def equals(that: Any): Boolean =
-      that.isInstanceOf[Date] && {
-        val o = that.asInstanceOf[Date]
-        o.day == day && o.month == month && o.year == year
-      }
+{% tabs equals-definition class=tabs-scala-version %}
+
+{% tab 'Scala 2' for=equals-definition %}
+```scala
+class Date(y: Int, m: Int, d: Int) extends Ord {
+  // previous decls here
+
+  override def equals(that: Any): Boolean =
+    that.isInstanceOf[Date] && {
+      val o = that.asInstanceOf[Date]
+      o.day == day && o.month == month && o.year == year
```

Comment:
Should we use pattern matching instead?

Reply:
Eg

```suggestion
    that match {
      case date: Date => date.day == day && date.month == month && date.year == year
      case _ => false
    }
```

Reply:
im not a fan of keeping this section, but ill make the change

Reply:
I kept the comparison of `instanceof` and `isInstanceOf`, but instead say it is not idiomatic to use instance of in Scala.